### PR TITLE
[Merged by Bors] - Parameterize stacks and migrate logging and monitoring stack

### PR DESCRIPTION
**Pull request by sbernauer**
## Description

* Demos and stacks can now be parametrized.
* Monitoring and logging stack added as well as parametrized.
* All the other stacks and demos are migrated to v2, but are *not* parametrized yet to reduce the scope of this PR

## Review Checklist
- [x] Code contains useful comments
- [x] (Integration-)Test cases added (or not applicable)
- [x] Documentation added (or not applicable)
- [x] Changelog updated (or not applicable)
- [x] Cargo.toml only contains references to git tags (not specific commits or branches)

Once the review is done, comment `bors r+` (or `bors merge`) to merge. [Further information](https://bors.tech/documentation/getting-started/#reviewing-pull-requests)

### CODE REVIEW

File: src/stack.rs [hunk 1]
```diff
@@ -43,6 +44,13 @@ pub enum CliCommandStack {
         #[arg(required = true, value_hint = ValueHint::Other)]
         stack: String,
 
+        /// List of parameters to use when installing the stack.
+        /// Every parameter need to have the format `<parameter>=<value>`, e.g. `adminPassword=secret123`.
+        /// Multiple parameters can be specified.
+        /// Use `stackable stack describe <stack>` to list the available parameters.
+        #[arg(short, long)]
+        parameters: Vec<String>,
```
Comment (maltesander) [inline src/stack.rs:52]:
```suggestion
        /// List of parameters to use when installing the stack.
        /// All parameters need to have the format `<parameter>=<value>`, e.g. `adminPassword=secret123`.
        /// Multiple parameters can be specified.
        /// Use `stackable stack describe <stack>` to list the available parameters.
        #[arg(short, long)]
        parameters: Vec<String>,
```

File: src/stack.rs [hunk 2]
```diff
@@ -96,31 +105,80 @@ struct Stacks {
...
+    ) -> Result<HashMap<String, String>, Box<dyn Error>> {
+        let mut parameters: HashMap<String, String> = stack_parameters
+            .iter()
+            .map(|p| (p.name.clone(), p.default.clone()))
+            .collect();
+
+        for parameter in cli_parameters {
+            let mut split = parameter.split('=');
+            let name = split.next().ok_or("Can not parse parameter. Parameters need to have the format `<parameter>=<value>`, e.g. `adminPassword=secret123`.")?;
+            let value = split.next().ok_or("Can not parse parameter. Parameters need to have the format `<parameter>=<value>`, e.g. `adminPassword=secret123`.")?;
```
Comment (maltesander) [inline src/stack.rs:162]: Could be a const?

File: src/templating.rs [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+                .map(|_| {
+                    let idx = rng.gen_range(0..PASSWORD_CHARSET.len());
+                    PASSWORD_CHARSET[idx] as char
+                })
+                .collect();
+
+            Ok(password.into())
+        },
+    )
+}
```
Comment (maltesander) [inline src/templating.rs:34]: Optional: Could use a crate.

**Review by maltesander — CHANGES_REQUESTED** [review]
LGTM overall, just some nit picking. Did not test yet.

File: src/demo.rs [hunk 4]
```diff
@@ -48,6 +49,20 @@ pub enum CliCommandDemo {
...
+        /// Use `stackable stack describe <stack>` to list the available parameters.
+        #[arg(short, long)]
+        stack_parameters: Vec<String>,
+
+        /// List of parameters to use when installing the demo.
+        /// Every parameter need to have the format `<parameter>=<value>`, e.g. `adminPassword=secret123`.
+        /// Multiple parameters can be specified.
+        /// Use `stackable demo describe <stack>` to list the available parameters.
+        #[arg(short, long)]
+        parameters: Vec<String>,
```
[the remark this answers is not in the document]
Comment (sbernauer) [reply src/demo.rs:64]: Was my first though as well.
My second was that "stack install -p foo=bar" and "demo install -p foo=bar" should look the same, hence calling the parameters only --parameters. The only exception is the parameters of the stack when installing the demo, where the name --parameters clashes

**Review by maltesander — COMMENTED** [review]
Logging Demo and Monitoring Stack work fine! Setting parameters works.
Changelog is missing.
LGTM otherwise.

File: src/demo.rs [hunk 5]
```diff
@@ -18,7 +18,8 @@ use std::{error::Error, ops::Deref, sync::Mutex};
 
 lazy_static! {
     pub static ref DEMO_FILES: Mutex<Vec<String>> = Mutex::new(vec![
-        "https://raw.githubusercontent.com/stackabletech/stackablectl/main/demos/demos-v1.yaml"
+        // "https://raw.githubusercontent.com/stackabletech/stackablectl/main/demos/demos-v2.yaml"
+        "demos/demos-v2.yaml"
```
Comment (sbernauer) [inline src/demo.rs:22]:
```suggestion
        "https://raw.githubusercontent.com/stackabletech/stackablectl/main/demos/demos-v2.yaml"
```

File: src/stack.rs [hunk 6]
```diff
@@ -9,11 +9,12 @@ use indexmap::IndexMap;
...
 use log::{debug, info, warn};
 use serde::{Deserialize, Serialize};
-use std::{error::Error, ops::Deref, sync::Mutex};
+use std::{collections::HashMap, error::Error, ops::Deref, sync::Mutex};
 
 lazy_static! {
     pub static ref STACK_FILES: Mutex<Vec<String>> = Mutex::new(vec![
-        "https://raw.githubusercontent.com/stackabletech/stackablectl/main/stacks/stacks-v1.yaml"
+//        "https://raw.githubusercontent.com/stackabletech/stackablectl/main/stacks/stacks-v2.yaml"
+        "stacks/stacks-v2.yaml"
```
Comment (sbernauer) [inline src/stack.rs:17]:
```suggestion
       "https://raw.githubusercontent.com/stackabletech/stackablectl/main/stacks/stacks-v2.yaml"
```

File: src/stack.rs (hunk 2 above)
Comment (sbernauer) [reply src/stack.rs]: cdd1a40aea9cc32e93b58aec11cb16eb2518f382

File: src/templating.rs (hunk 3 above)
Comment (sbernauer) [reply src/templating.rs]: aec4888c8bac0bf28b4dbe34516b3ac0dacb025d

**Review by maltesander — APPROVED** [review]
LGTM.

**Pull request closed without merging**